server/routerlicious/packages/services-utils/src/redisClientConnectionManager.ts

```diff
@@ -32,7 +32,7 @@ export class RedisClientConnectionManager implements IRedisClientConnectionManag
 	};
 
 	constructor(
-		redisOptions?: Redis.RedisOptions,
+		redisOptions?: Redis.RedisOptions & Redis.ClusterOptions,
```

Comment:
So technically no in this case, with the reason being that every property in RedisOptions and ClusterOptions is optional. If this were not the case, then an object of type (RedisOptions & ClusterOptions) would need to have all required props of both. 

Just to make sure that the constructor can always take either RedisOptions, ClusterOptions, or (RedisOptions & ClusterOptions) though, I changed the type in the constructor/field to Partial< RedisOptions & ClusterOptions>

I also checked through FF repo, and only found one caller that passes in redisOptions (except a few which use ioredis-mock) and I switched the type of that one to be the & type, just incase someone in the future wants to add some clustering specific options there (although, I think in most cases we would want to add these how I added clusterRetryStrategy in this PR, instead of adding it in each redisOptions object).